v2/internal/pkg/operator/filtered_collector.go

```diff
@@ -273,7 +273,17 @@ func (o FilterCollector) collectOperator( //nolint:cyclop // TODO: this needs fu
 		rebuiltTag = tag
 	}
 
+	// OCPBUGS-52470
+	// check if the original operator was mirrored by digest
 	componentName := imgSpec.ComponentName() + "." + result.Digest
+	if imgSpec.IsImageByDigestOnly() && o.Opts.IsMirrorToDisk() {
+		tag, err := digestOfFilter(op)
```

Comment:
Hi @lmzuccarelli 
Just for my understanding, we are using the digestOfFilter instead of the digest as a tag (which we've adopted in v2 for any images that don't have tags), because the digest as a tag is already used for the unfiltered catalog image when we save it to the cache. Is that right?

Reply:
@sherine-k - yes correct